cibyl/sources/elasticsearch/client.py

```diff
@@ -37,19 +37,36 @@ def __init__(self: object,
         :param port: Port for sending REST requests to the instance
         :type port: str, optional"""
         self.address = f"{host}:{port}"
+        self.connection = None
 
-    def connect(self: object) -> Elasticsearch:
+    def connect(self) -> Elasticsearch:
         """Connects to the elasticsearch instance
 
         :return: Elasticsearch module instance
         :rtype: Elasticsearch
         :raises: ElasticSearchError:
                  If exists an unhandled connection error
         """
-        es_client = Elasticsearch(self.address)
-        if not es_client.ping():
-            raise ElasticSearchError(f"Error connecting to "
-                                     f"Elasticsearch: {self.address}")
+        self.connection = Elasticsearch(self.address)
```

Comment:
If a connection is already open, you should close it before opening a new one.

Reply:
I don't get this. Do you mean to check something like this?
```
if self.connection
    self.disconnect()
self.connection = Elasticsearch(self.address)
```

Reply:
Sure, just if you want to be extra safe.

Reply:
You're right. I added it.